IdCardManager.js

```diff
@@ -1,125 +1,145 @@
-        [LANGUAGE_RU]: 'Отсутствует необходимое программное обеспечение'
+        [LANGUAGE_ET]: "Vajalik tarkvara on puudu",
+        [LANGUAGE_EN]: "Unable to find software",
+        [LANGUAGE_LT]: "Nerasta programinės įranga",
+        [LANGUAGE_RU]: "Отсутствует необходимое программное обеспечение",
     },
 
     technical_error: {
-        [LANGUAGE_ET]: 'Tehniline viga',
-        [LANGUAGE_EN]: 'Technical error',
-        [LANGUAGE_LT]: 'Techninė klaida',
-        [LANGUAGE_RU]: 'Техническая ошибка'
+        [LANGUAGE_ET]: "Tehniline viga",
+        [LANGUAGE_EN]: "Technical error",
+        [LANGUAGE_LT]: "Techninė klaida",
+        [LANGUAGE_RU]: "Техническая ошибка",
     },
 
     not_allowed: {
-        [LANGUAGE_ET]: 'Veebis allkirjastamise käivitamine on võimalik vaid https aadressilt',
-        [LANGUAGE_EN]: 'Web signing is allowed only from https:// URL',
-        [LANGUAGE_LT]: 'Web signing is allowed only from https:// URL',
-        [LANGUAGE_RU]: 'Подпись в интернете возможна только с URL-ов, начинающихся с https://'
+        [LANGUAGE_ET]:
+            "Veebis allkirjastamise käivitamine on võimalik vaid https aadressilt",
+        [LANGUAGE_EN]: "Web signing is allowed only from https:// URL",
+        [LANGUAGE_LT]: "Web signing is allowed only from https:// URL",
+        [LANGUAGE_RU]:
+            "Подпись в интернете возможна только с URL-ов, начинающихся с https://",
     },
 };
 
-
 class IdCardManager {
     constructor(language) {
         this.language = language || LANGUAGE_ET;
 
-        this._cert = null;
+        this.certificate = null;
     }
 
     initializeIdCard() {
         return new Promise(function (resolve, reject) {
             if (window.hwcrypto.use("auto")) {
                 resolve();
-            }
-
-            else {
+            } else {
                 reject("Backend selection failed");
             }
         });
     }
 
+    /**
+     * Calls `window.hwcrypto.getCertificate()` with the selected language.
+     * Resolves to a HEX-encoded certificate. The certificate is to be embedded into the XML signature
+     * that is generated on the backend.
+     *
+     * `hwcrypto.getCertificate()` resolves to a certificate object ("handle"):
+     * {
+     *      encoded: UInt8Array, raw certificate
+     *      hex: String, HEX-encoded certificate
+     * }
+     *
+     * The "handle" is to be used in the sign() call. So we also store it on the instance.
+     *
+     * https://github.com/hwcrypto/hwcrypto.js/wiki/APIv2#getcertificate
+     *
+     * @returns {Promise<String>}
+     */
     getCertificate() {
         return new Promise((resolve, reject) => {
-            let lParam = {lang: this.language};
-
-            window.hwcrypto.getCertificate(lParam).then((rCert) => {
-                this._cert = rCert;
-                resolve();
-            },
-
-            (err) => {
-                reject(err);
-            });
+            let lParam = { lang: this.language };
```

Comment:
I guess this can be a const too

Reply:
I pretty much didn't touch it; but yeah makes sense to make it const :)